feature/experimental/gribi/otg_tests/programming_with_reload/programming_with_reload_test.go

```diff
@@ -0,0 +1,513 @@
+			gnmi.Update(t, args.dut, dc.Interface(p.Name()).Config(), i)
+		}); errMsg != nil {
+			t.Logf("Got testt.CaptureFatal errMsg: %s, retrying ...", *errMsg)
+		} else {
+			break
+		}
+	}
+}
+
+// addStaticRoute configures static route needed for decap path
+func addStaticRoute(t *testing.T, dut *ondatra.DUTDevice, ip string) {
+	d := gnmi.OC()
+	s := &oc.Root{}
+	static := s.GetOrCreateNetworkInstance(deviations.DefaultNetworkInstance(dut)).GetOrCreateProtocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_STATIC, deviations.StaticProtocolName(dut))
+	ipv4Nh := static.GetOrCreateStatic(innerDstIP1 + "/" + mask).GetOrCreateNextHop("0")
+	ipv4Nh.NextHop, _ = ipv4Nh.To_NetworkInstance_Protocol_Static_NextHop_NextHop_Union(ip)
+	gnmi.Update(t, dut, d.NetworkInstance(deviations.DefaultNetworkInstance(dut)).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_STATIC, deviations.StaticProtocolName(dut)).Config(), static)
+}
+
+// Reload chassis
+func (args *testArgs) reloadChassis(t *testing.T) {
+
+	gnoiClient, err := args.dut.RawAPIs().BindingDUT().DialGNOI(context.Background())
+	if err != nil {
+		t.Fatalf("Error dialing gNOI: %v", err)
+	}
+	bootTimeBeforeReboot := gnmi.Get(t, args.dut, gnmi.OC().System().BootTime().State())
+	t.Logf("DUT boot time before reboot: %v", bootTimeBeforeReboot)
+
+	rebootRequest := &spb.RebootRequest{
+		Method:  spb.RebootMethod_COLD,
+		Delay:   0,
+		Message: "Reboot chassis with delay",
+		Force:   true,
+	}
+
+	t.Logf("Send reboot request: %v", rebootRequest)
+	rebootResponse, err := gnoiClient.System().Reboot(context.Background(), rebootRequest)
+	t.Logf("Got reboot response: %v, err: %v", rebootResponse, err)
+	if err != nil {
+		t.Fatalf("Failed to reboot chassis with unexpected err: %v", err)
+	}
+
+	// adding sleep time for reload to begin
+	time.Sleep(1 * time.Minute)
+
+	// establish new gribi programming following reload
+	err_msg := args.gribi_reconnect(t)
+	if err_msg != nil {
+		t.Fatalf("Failure: %d", err_msg)
+	}
+}
+
+// gribi reconnect following reload
+func (args *testArgs) gribi_reconnect(t *testing.T) error {
+
+	reboot_timer := time.Now()
+	client := gribi.Client{
+		DUT:         args.dut,
+		FIBACK:      true,
+		Persistence: true,
+	}
+
+	for {
+		// following reboot, if gribi connection takes longer than maxRebootTime return failure error msg
+		if time.Since(reboot_timer).Seconds() > maxRebootTime {
+			return fmt.Errorf("Device took longer than %d to reboot", maxRebootTime)
+		} else {
+			// Attempt starting gribi client session and check the error code,
+			// if its context deadline exceeded DUT is still rebooting and try again
+			if err := client.Start(t); err != nil {
+				t.Logf("gRIBI Connection could not be established because device is still rebooting: %v\nRetrying...", err)
+				if errors.Is(err, context.DeadlineExceeded) {
+					time.Sleep(1 * time.Second)
+				} else {
+					// if error is UNAVAIABLE, try creating gribi session for maxGribiConnectTime
```

Comment:
You may want to check the specific error code here (UNAVAILABLE) vs just any other error?
something like

if (context.DeadlineExceeded)
else if (UNAVAILABLE)
else (received an unexpected gRPC error)

Reply:
Ack